btree/btree.c

```diff
@@ -1828,15 +1892,17 @@ static int64_t node_alloc(struct node_op *op, struct td *tree, int size,
 	return nxt_state;
 }
 
-#ifndef __KERNEL__
 static int64_t node_free(struct node_op *op, struct nd *node,
 			 struct m0_be_tx *tx, int nxt)
 {
 	int shift = node->n_type->nt_shift(node);
-	node->n_ref--;
+
+	node_refcnt_update(node, false);
 	node->n_type->nt_fini(node);
 
 	if (node->n_ref == 0) {
+		ndlist_tlink_del_fini(node);
+		m0_rwlock_fini(&node->n_lock);
```

Comment:
after reaching to this point, some other thread may request same  node, and increase n_ref . Such scenarios needs to be think. node_get and node_free should be in sync

Reply:
Moved `n_delayed_free = true` to top of the function, so that race condition is handled.